tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -938,7 +943,12 @@ public void onContainersCompleted(List<ContainerStatus> completedContainers) {
           if (task.getSessionId() != session.sessionId) {
             return;
           }
-          // Update TensorFlowSession on the state of the task.
+
+          if (task.getTaskIndex().equals("0") && task.getJobName().equals(WORKER_JOB_NAME)) {
+            chiefWorkerFailed = true;
```

Comment:
Not sure I follow this part. Won't this mark it failed no matter if the chief succeeds or fails?